workflow-controller/controller-service/src/main/java/io/hyscale/controller/processor/ServiceSpecProcessor.java

```diff
@@ -0,0 +1,185 @@
+/**
+ * Copyright 2019 Pramati Prism, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.hyscale.controller.processor;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.AbstractMap.SimpleEntry;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.hyscale.commons.exception.HyscaleException;
+import io.hyscale.commons.io.HyscaleFilesUtil;
+import io.hyscale.commons.logger.WorkflowLogger;
+import io.hyscale.commons.models.Status;
+import io.hyscale.commons.utils.ObjectMapperFactory;
+import io.hyscale.controller.activity.ControllerActivity;
+import io.hyscale.controller.constants.WorkflowConstants;
+import io.hyscale.controller.exception.ControllerErrorCodes;
+import io.hyscale.controller.model.EffectiveServiceSpec;
+import io.hyscale.controller.model.HyscaleInputSpec;
+import io.hyscale.controller.util.ServiceProfileUtil;
+import io.hyscale.controller.util.ServiceSpecUtil;
+import io.hyscale.generator.services.model.ServiceMetadata;
+import io.hyscale.servicespec.commons.builder.EffectiveServiceSpecBuilder;
+import io.hyscale.servicespec.commons.builder.MapFieldDataProvider;
+import io.hyscale.servicespec.commons.builder.ServiceInputType;
+import io.hyscale.servicespec.commons.model.service.ServiceSpec;
+
+@Component
+public class ServiceSpecProcessor {
```

Comment:
Something close to EffectiveServiceSpec ?
The contract is mostly get and merge of it

Reply:
We have a EffectiveServiceSpecBuilder which takes care of merging each service spec with profile. I could change it to ServiceSpecMapper as its mostly mapping service spec and profile then calling EffectiveServiceSpecBuilder?